rosbag2_transport/test/rosbag2_transport/test_qos.cpp

```diff
@@ -19,20 +19,21 @@
 
 #include "rmw/types.h"
 
-#include "rosbag2_transport/qos.hpp"
+#include "rosbag2_storage/qos.hpp"
 
 TEST(TestQoS, serialization)
 {
-  rosbag2_transport::Rosbag2QoS expected_qos;
+  rosbag2_storage::Rosbag2QoS expected_qos;
   YAML::Node offered_qos_profiles;
   offered_qos_profiles.push_back(expected_qos);
 
   std::string serialized = YAML::Dump(offered_qos_profiles);
   YAML::Node loaded_node = YAML::Load(serialized);
-  auto deserialized_profiles = loaded_node.as<std::vector<rosbag2_transport::Rosbag2QoS>>();
+  auto deserialized_profiles = YAML::decode_for_version<std::vector<rosbag2_storage::Rosbag2QoS>>(
+    loaded_node, 9);
   ASSERT_EQ(deserialized_profiles.size(), 1u);
```

Comment:
It would be nice to add a test case when version is equal to 8 and we are trying to decode a new format.